packages/react-core/src/components/ApplicationLauncher/__tests__/ApplicationLauncher.test.tsx

```diff
@@ -55,8 +55,15 @@ describe('ApplicationLauncher', () => {
 
   test('custom icon', () => {
     const { asFragment } = render(
-      <ApplicationLauncher items={dropdownItems} isOpen toggleIcon={<HelpIcon id="test-icon" />} />
+      <React.StrictMode>
+        <ApplicationLauncher items={dropdownItems} isOpen toggleIcon={<HelpIcon id="test-icon" />} />
+      </React.StrictMode>
     );
     expect(asFragment()).toMatchSnapshot();
   });
+
+  test('Renders in strict mode', () => {
+    const { asFragment } = render(<ApplicationLauncher removeFindDomNode isOpen items={dropdownItems} />);
```

Comment:
I think there is a way here to test for console warnings/error - so you could test that there is or is not console output in this situation. @wise-king-sullyman @Dominik-Petrik do you know how to do this?

Reply:
Yes great call! You can test for that behavior with something like:

```js
test('Name', () => {
  const consoleError = jest.spyOn(console, 'error').mockImplementation();

  render(
    <Component />
  );

  expect(consoleError).not.toHaveBeenCalled();
  ```

Reply:
I think mocking console.warn function could be the way.
`console.warn = jest.fn()

//render

expect(console.warn).toBeCalled()`

Should fail when there is no console warning.

What do you think @wise-king-sullyman?

Reply:
I think mocking console.warn function could be the way.
`console.warn = jest.fn()`
`//render`
`expect(console.warn).toBeCalled()`

Should fail when there is no console warning.

What do you think @wise-king-sullyman?

Reply:
@Dominik-Petrik I'm not 100% sure, but I _think_ that setting consoles methods to be a mock is generally not the preferred approach because of it's potential to cause side effects in other tests if the mock isn't restored.

From my understanding using `jest.spyOn` avoids this problem, and so it's generally the preferred approach.

That being said you actually sent me down a little bit of a rabbit hole and I learned that using `.mockImplementation` causes the spy to cause side effects if the original functionality isn't restored as well! For that reason I'm editing what I originally proposed, and we should probably remove where `.mockImplementation` is used on spies, or at least re-evaluate if it's really needed where it's being used and ensure that we restore the original behavior if so! I'll make an issue for that 🙂